### ISSUE TITLE
Sub-commands

### ISSUE THREAD

**Issue by The-Compiler** [The-Compiler/qutebrowser#1095]
For certain commands (session-* tab-* view-* yank-* zoom-* download-*) it would make more sense if there are sub-commands, like `:download cancel` (think `git`).

This would need some adjustments for the completion and doc generation first though. It certainly would fit into the "reduce amount of commands" philosophy of qutebrowser.

**Comment by rcorre** [comment]
In the interest of reducing the amount of commands, how about `completion-item-next` and `completion-item-next -reverse` instead of `completion-item-next` and `completion-item-previous`? Internally these are both one-liners that pass a different flag to the helper function.

**Comment by lahwaacz** [comment]
Why not `completion-item --next` and `completion-item --previous`? That's almost like subcommands, but without the difficulties.

**Comment by rcorre** [comment]
I thought about that, but was bothered by the fact that, on its own, `completion-item` doesn't sound like a command.

**Comment by The-Compiler** [comment]
Hmm, not sure - isn't `:completion-item-next --reverse` a bit weird?

**Comment by rcorre** [comment]
Yeah, maybe. Everything sounds weird to me now. I guess `completion-item --next/prev` isn't terrible.

**Comment by The-Compiler** [comment]
Looking at #1567, what about `:completion-focus {next,prev}` (as positional argument, not flag), and then we can add `next-category` and `prev-category` later?

**Comment by rcorre** [comment]
I thought about `select` but that sounded more like I was commiting to the currently chosen element. I like `focus`, and I think the positional argument will be nice once we have completion for `choices` (hopefully coming soon, now that tests are in place).

**Comment by The-Compiler** [comment]
@rcorre Do you want to work on a `:completion-focus` command, or should I open a separate issue so we don't forget about it?

**Comment by The-Compiler** [comment]
For another low-hanging fruit, there's also `:search-next` and `:search-prev`. I guess they could be merged into `:search` as flags (as `:search` without argument already clears the search term), but I'm not sure.

**Comment by pkill-nine** [comment]
If subcommands are made, could they also be searched by the completion? I.e. typing `clear` suggesting the command `download clear`?

**Comment by The-Compiler** [comment]
That'd certainly be possible, though I'm not sure if it'd be desirable.
